### PR TITLE
33 statistics domain

### PR DESCRIPTION
**Pull request by renan-cunha**

### CODE REVIEW

File: mobile/lib/features/statistics/domain/use_cases/get_number_of_correct_words_read_per_minute_use_case.dart [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  });
+
+  @override
+  Future<Either<Failure, double>> call(AudioParams params) async {
+    final output = await getTextUseCase(params.audio.textId);
+    int number_words;
+    int audioMinutes;
+    if (output.isRight()) {
+      final MyText text = output.getOrElse(() => null);
+      number_words = text.body.split(" ").length;
```
Comment (VitorCantao) [inline mobile/lib/features/statistics/domain/use_cases/get_number_of_correct_words_read_per_minute_use_case.dart:35]: I think this should be in the `MyText` entity, probably with a `get`ter

Comment (VitorCantao) [reply mobile/lib/features/statistics/domain/use_cases/get_number_of_correct_words_read_per_minute_use_case.dart:35]: `wordCount` and `words`

File: mobile/lib/features/statistics/domain/use_cases/get_number_of_correct_words_read_per_minute_use_case.dart [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  GetNumberOfCorrectWordsReadPerMinuteUseCase({
+    @required this.getTextUseCase,
+    @required this.getAudioDurationInMinutesUseCase,
+    @required this.getNumberOfMistakenWordsUseCase,
+    @required this.getCorrectionFromIdUseCase,
+  });
+
+  @override
+  Future<Either<Failure, double>> call(AudioParams params) async {
```
Comment (VitorCantao) [inline mobile/lib/features/statistics/domain/use_cases/get_number_of_correct_words_read_per_minute_use_case.dart:29]: We can probably extract some methods inside this method to make it cleaner and clear

File: mobile/lib/features/statistics/domain/use_cases/get_number_of_mistaken_words_use_case.dart [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import 'package:dartz/dartz.dart';
+import 'package:mobile/core/error/failures.dart';
+import 'package:mobile/core/use_cases/use_case.dart';
+import 'package:mobile/features/text_correction/domain/entities/correction.dart';
+import 'package:mobile/features/text_correction/domain/use_cases/correction_params.dart';
+
+class GetNumberOfMistakenWordsUseCase
+    implements UseCase<int, CorrectionParams> {
+  @override
+  Future<Either<Failure, int>> call(CorrectionParams params) async {
```
Comment (VitorCantao) [inline mobile/lib/features/statistics/domain/use_cases/get_number_of_mistaken_words_use_case.dart:10]: Same comment I made in the other one

File: mobile/test/core/data/text_database_test.dart [hunk 4]
```diff
@@ -187,7 +185,41 @@ void main() {
...
+          throwsA(const TypeMatcher<SqliteException>()));
+    });
+
+    test("should return a text", () async {
+      final result = await database.getText(1);
+      expect(result, tValidTextModel1);
+    });
+
+    /*
+    test("should return a list with one text", () async {
```
Comment (VitorCantao) [inline mobile/test/core/data/text_database_test.dart:206]: Commented code should be deleted

**Review by VitorCantao — CHANGES_REQUESTED** [review]

File: mobile/lib/features/statistics/domain/use_cases/get_audio_duration_in_seconds_use_case.dart [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import 'package:dartz/dartz.dart';
+import 'package:mobile/core/error/failures.dart';
+import 'package:mobile/core/use_cases/use_case.dart';
+import 'package:mobile/features/audio_management/domain/entities/audio.dart';
+import 'package:mobile/features/audio_management/domain/use_cases/audio_params.dart';
+
+class GetAudioDurationInMinutesUseCase implements UseCase<int, AudioParams> {
```
[the remark this answers is not in the document]
Comment (renan-cunha) [reply mobile/lib/features/statistics/domain/use_cases/get_audio_duration_in_seconds_use_case.dart:7]: this use isn't finished, and is not returning the actual minutes now. Flutter audio management is different than other frameworks. I will wait for Canavarro to test with an actual audio so I can finish this implementation.

**Review by VitorCantao — APPROVED** [review]

**Pull request merged**